examples/error.rs

```diff
@@ -37,17 +37,63 @@ impl Handler for ErrorHandler {
 
 impl BeforeMiddleware for ErrorProducer {
     fn before(&self, _: &mut Request) -> IronResult<()> {
-        Err(IronError::new(StringError("Error".to_string()), status::BadRequest))
+        Err(IronError::new(StringError("Error you cannot access the resource!".to_string()), status::BadRequest))
     }
 }
 
+
+// to handle the Error message by using an AroundMiddleware
+struct LogEnabler;
+
+struct HandlerWithLog<H: Handler> {
+    handler: H
+}
+
+impl AroundMiddleware for LogEnabler {
+
+    fn around(self, handler: Box<Handler>) -> Box<Handler> {
+        Box::new(HandlerWithLog { handler: handler } ) as Box<Handler>
+    }
+}
+ 
+impl <H: Handler> Handler for HandlerWithLog<H> {
+
+    fn handle(&self, req: &mut Request) -> IronResult<Response> {
+    
+        let res = self.handler.handle(req);
+        
+        {
+          match res {
+            Ok(_) => panic!("is not possible!"),
+            Err(IronError {error: ref what_went_wrong, ref response }) => println!("what went wrong: {:?} - response: {:?}", what_went_wrong, response)
```

Comment:
if you just do `IronError { ref error, ref response }` I think that will be clearer